controllers/helmrelease_controller.go

```diff
@@ -693,12 +699,22 @@ func (r *HelmReleaseReconciler) garbageCollectHelmRelease(logger logr.Logger, hr
 func (r *HelmReleaseReconciler) handleHelmActionResult(hr *v2.HelmRelease, revision string, err error, action string, condition string, succeededReason string, failedReason string) error {
 	if err != nil {
 		msg := fmt.Sprintf("Helm %s failed: %s", action, err.Error())
-		v2.SetHelmReleaseCondition(hr, condition, corev1.ConditionFalse, failedReason, msg)
+		apimeta.SetStatusCondition(&hr.Status.Conditions, metav1.Condition{
+			Type:    condition,
+			Status:  metav1.ConditionFalse,
+			Reason:  failedReason,
+			Message: msg,
+		})
 		r.event(*hr, revision, events.EventSeverityError, msg)
 		return &ConditionError{Reason: failedReason, Err: errors.New(msg)}
 	} else {
 		msg := fmt.Sprintf("Helm %s succeeded", action)
-		v2.SetHelmReleaseCondition(hr, condition, corev1.ConditionTrue, succeededReason, msg)
+		apimeta.SetStatusCondition(&hr.Status.Conditions, metav1.Condition{
+			Type:    condition,
+			Status:  metav1.ConditionTrue,
+			Reason:  succeededReason,
+			Message: msg,
+		})
```

Comment:
Don't we want to use https://github.com/fluxcd/pkg/pull/49 here so that we will get the observed generation handling once it is implemented?